ts/components/ButtonDefaultOpacity.tsx

```diff
@@ -1,14 +1,42 @@
 import { Button, NativeBase } from "native-base";
 import * as React from "react";
+import customVariables from "../theme/variables";
 
 const defaultActiveOpacity = 1.0;
+
+type Props = NativeBase.Button & React.Props<Button>;
+
+/**
+ *
+ * @param height
+ */
+const calculateSlop = (height: number): number => {
```

Comment:
To avoid compute slop on each button spawn, could we compute that values only a time?

```typescript
const smallSlop = calculateSlop(customVariables.btnSmallHeight);
const xsmallSlop = calculateSlop(customVariables.btnXSmallHeight);
const dafaultSlop = calculateSlop(customVariables.btnHeight);
/**
 * This is a temporary solution to extend the touchable area using the existing theme system.
 * @deprecated
 * @param props
 */
const getSlopForCurrentButton = (props: Props) => {
  if (props.small) {
    return smallSlop;
  } else if (props.xsmall) {
    return xsmallSlop;
  }
  return dafaultSlop;
};
```

Reply:
yep, definitely better :D changed in 640746a